includes/polling/applications/sagan.inc.php

```diff
@@ -78,7 +78,9 @@
     $tags = ['name' => $name, 'app_id' => $app->app_id, 'rrd_def' => $rrd_def, 'rrd_name' => $rrd_name];
     data_update($device, 'app', $tags, $fields);
 }
-$old_instances = $app->app['instances'];
+
+$old_instances = $app->app['instances'] ?? [];
+$new_data['instances'] = $instances;
```

Comment:
$new_data is an un-initialized variable here.
It is better to do this
$new_data = ['instances' => $instances];

Additionally, it might be better to simply set the instances and use $app->getOriginal() and $app->isDirty()

Reply:
@murrant Aweosme. Fixed. :)